### DISCUSSION TITLE
How to edit wsl.conf/disable resolve.conf generation

### DISCUSSION THREAD

**Discussion by shaqaruden** [nix-community/NixOS-WSL#379] [Q&A]
How would I go about disabling auto generation of resolve.conf? I have tried to just edit `/etc/wsl.conf` but this file is readonly even with sudo. Do I just change the permissions of that file or is there another way?

**Comment by nzbr** [comment]
You can safely set the file to writeable (`644`) and then edit it with sudo. It gets set to readonly due to how the tarballs are built.
I created #384 to have the file be writeable by default in the future.

The config option you're looking for is probably `wsl.wslConf.network.generateResolvConf`

**Reply by deng232** [reply]
```/etc/wsl.conf``` was actually a symlink link to actual file in ```/nix/store.```, and ```chmod``` I have tried but somehow doesn't work there. 
The problem I have is wsl keep generated an unusable nameserver for me so I can't upate channel and doing rebuild switch.

**Reply by nzbr** [reply]
If you really have to edit `wsl.conf` manually, you could `rm` the symlink and copy the file from the nix store there, so that you can edit it.
What I'd recommend instead is setting the option through `wsl.wslConf` and manually override the nameservers once so that you can rebuild. To do that, run the following in a root shell:
```
rm /etc/resolv.conf
echo "nameserver 0.0.0.0" > /etc/resolv.conf
```
replace `0.0.0.0` with the IP address of the DNS server you want to use
